### ISSUE TITLE
Error in FTP-Upload

### ISSUE THREAD

**Issue by marioschaefer** [tabernarious/f5-automated-backup-iapp#6]
Hi,

the FTP-Upload fails.
This is because the name used for generating the filename in the  the shellscript is "${fname}BACKUPFILESCRIPTEXTENSION" but fname already includes the backupextension.
So the script ends up trying to upload "backupdir/filename.extension.extension"

Fix:
just change "${fname}BACKUPFILESCRIPTEXTENSION" to "${fname}":
```tcl
# Updated command v2.2.4 to force binary transfer.
exec echo -e "ftp_function()\n{\n\tf5masterkey=\$(f5mku -K)\n\tusername=\$(echo \"ENCRYPTEDUSERNAME\" | openssl aes-256-ecb -salt -a -A -d -k \${f5masterkey})\n\tpassword=\$(echo \"ENCRYPTEDPASSWORD\" | openssl aes-256-ecb -salt -a -A -d -k \${f5masterkey})\n\tserver=\$(echo \"ENCRYPTEDSERVER\" | openssl aes-256-ecb -salt -a -A -d -k \${f5masterkey})\n\tdirectory=\$(echo \"ENCRYPTEDDIRECTORY\" | openssl aes-256-ecb -salt -a -A -d -k \${f5masterkey})\n\n\tftp_return=\$(ftp -n \${server} << END_FTP\nquote USER \${username}\nquote PASS \${password}\nbinary\nput BACKUPDIRECTORY/${fname} \${directory}/${fname}\nquit\nEND_FTP\n)\n\tif \[ \"\$ftp_return\" == \"\" \]\n\tthen\n\t\treturn 0\n\telse\n\t\techo \"\$ftp_return\" >> /var/tmp/scriptd.out\n\t\treturn 1\n\tfi\n}\n\nftp_function" > $scriptfile
# Original command which allowed ascii transfer.
#exec echo -e "put()\n{\n\tfields=\"username password server directory\"\n\ti=1\n\tf5masterkey=\$(f5mku -K)\n\tfor current_field in \$fields ; do\n\t\tsedcommand=\"\${i}p\"\n\t\tcurrent_encrypted_value=\$(sed -n \"\$sedcommand\" $configfile)\n\t\tcurrent_decrypted_value=\$(echo \"\$current_encrypted_value\" | openssl aes-256-ecb -salt -a -d -k \$f5masterkey)\n\t\teval \"\$current_field=\$current_decrypted_value\"\n\t\tlet i=\$i+1\n\t\tunset current_encrypted_value current_decrypted_value sedcommand\n\tdone\n\tftp -n \${server} << END_FTP\nquote USER \${username}\nquote PASS \${password}\nput BACKUPDIRECTORY/${fname} \${directory}/${fname}\nquit\nEND_FTP\n\treturn \$?\n}\n\nput" > $scriptfile
```tcl

**Comment by tabernarious** [comment]
Thanks! I'll try to update this later this week.

**Comment by tabernarious** [comment]
Fixed by @akhmarov in v3.1.1
